nectar/downloaders/threaded.py

```diff
@@ -273,119 +274,128 @@ def _fetch(self, request, session):
-            to failed_netlocs so that a new connection can be attempted.
-            """
-            _logger.warning("Request Timeout - Connection with {url} timed out.".format(
-                url=request.url)
-            )
-            report.download_connection_error()
-
-        except DownloadCancelled as e:
-            _logger.info(str(e))
-            report.download_canceled()
-
-        except DownloadFailed as e:
-            _logger.info('Download failed: %s' % str(e))
-            report.error_msg = e.args[2]
-            report.error_report['response_code'] = e.args[1]
-            report.error_report['response_msg'] = e.args[2]
-            report.download_failed()
-
-        except Exception as e:
-            _logger.exception(e)
-            report.error_msg = str(e)
-            report.download_failed()
+                if ignore_encoding or self.config.stream:
+                    chunks = self.chunk_generator(response.raw, self.buffer_size)
+                else:
+                    chunks = response.iter_content(self.buffer_size)
+
+                for chunk in chunks:
+                    if self.is_canceled or request.canceled:
+                        raise DownloadCancelled(request.url)
+
+                    file_handle.write(chunk)
+
+                    bytes_read = len(chunk)
+                    report.bytes_downloaded += bytes_read
+
+                    now = datetime.datetime.now()
+
+                    if now - last_update_time >= progress_interval:
+                        last_update_time = now
+                        self.fire_download_progress(report)
+
+                    with self._bytes_lock:
+                        if now - self._time_bytes_this_second_was_cleared >= ONE_SECOND:
+                            self._bytes_this_second = 0
+                            self._time_bytes_this_second_was_cleared = now
+                        self._bytes_this_second += bytes_read
+
+                    if max_speed is not None and self._bytes_this_second >= max_speed:
+                        # it's not worth doing fancier mathematics than this, very
+                        # fine-grained sleep times [1] are not honored by the system
+                        # [1] for example, sleeping the remaining fraction of time
+                        # before this second is up
+                        time.sleep(0.5)
+
+                # guarantee 1 report at the end
+                self.fire_download_progress(report)
+
+            except SkipLocation:
+                _logger.debug("Skipping {url} because {netloc} could not be reached.".format(
+                    url=request.url, netloc=netloc)
+                )
+                report.download_skipped()
+
+            except requests.ConnectionError as e:
+                # retry only if there's indication of connection reset
+                if nretry < DEFAULT_GENERIC_TRIES - 1 and not response.status_code:
```

Comment:
s/if nretry < DEFAULT_GENERIC_TRIES - 1 and not response.status_code:/if nretry < DEFAULT_GENERIC_TRIES - 1